rgbd/src/server_shm.cpp

```diff
@@ -120,6 +128,29 @@ void ServerSHM::send(const Image& image)
 
 // ----------------------------------------------------------------------------------------
 
+void ServerSHM::checkSHMThreadFunc(const float frequency)
+{
+    ROS_DEBUG("ServerSHM::checkSHMThreadFunc");
+    ros::Rate r(frequency);
+    while(nh_.ok())
+    {
+        ROS_DEBUG_STREAM("ServerSHM::checkSHMThreadFunc: checking shm on: " << shared_mem_name_);
+        try
+        {
+            ipc::shared_memory_object(ipc::open_only, shared_mem_name_.c_str(), ipc::read_only);
```

Comment:
We should note that this will only detect our problems in shared memory. We still need to be weary of what is causing the corruption in the first place

Reply:
True. Based on what I have seen up to now, it is not in our code.

I have seen it happen on `hero1` and `hero2` with just 1 second in between. Which could suggest it is something electrical. Or that timing was totally coincidence.
Any other case I have seen it happen, there was nothing in the logs, which indicates it is caused by our code. Which either leaves it to be the filesystem, posix or boost. But not something we can do there.